app/src/main/java/com/lagradost/cloudstream3/ui/player/AbstractPlayerFragment.kt

```diff
@@ -607,23 +617,57 @@ abstract class AbstractPlayerFragment(
     }
 
     fun nextResize() {
-        resizeMode = (resizeMode + 1) % PlayerResize.entries.size
-        resize(resizeMode, true)
+        showAspectRatioDialog()
     }
 
     fun resize(resize: Int, showToast: Boolean) {
         resize(PlayerResize.entries[resize], showToast)
     }
 
+    private fun showAspectRatioDialog() {
+        player.handleEvent(CSPlayerEvent.Pause, PlayerEventSource.UI)
+
+        val aspectRatioOptions = PlayerResize.values().map { getString(it.nameRes) }
+
+        activity?.showDialog(
+            aspectRatioOptions,
+            -1,
+            getString(R.string.choose_aspect_ratio),
+            false, // Immediate selection without an Apply button
+            dismissCallback = {
+                player.handleEvent(CSPlayerEvent.Play, PlayerEventSource.UI)
+            },
+            callback = { selectedIndex ->
+                val selectedOption = PlayerResize.values()[selectedIndex]
+                resize(selectedOption, true)
+                player.handleEvent(CSPlayerEvent.Play, PlayerEventSource.UI)
+            }
+        )
+    }
+
+
+
     @SuppressLint("UnsafeOptInUsageError")
     fun resize(resize: PlayerResize, showToast: Boolean) {
         DataStoreHelper.resizeMode = resize.ordinal
         val type = when (resize) {
             PlayerResize.Fill -> AspectRatioFrameLayout.RESIZE_MODE_FILL
             PlayerResize.Fit -> AspectRatioFrameLayout.RESIZE_MODE_FIT
             PlayerResize.Zoom -> AspectRatioFrameLayout.RESIZE_MODE_ZOOM
+            PlayerResize.SixteenByNine -> 16.0f/9.0f
+            PlayerResize.NineBySixteen -> 9.0f/16.0f
+            PlayerResize.OneByOne -> 1.0f/1.0f
+            PlayerResize.IMAX -> 1.43f/1.0f
+            PlayerResize.DolbyVision -> 2.39f/1.0f
+            PlayerResize.UltraWide -> 21.0f/9.0f
+            PlayerResize.Expanded -> 1.90f/1.0f
+        }
+        if(type is Float){
+            playerView?.resizeMode = AspectRatioFrameLayout.RESIZE_MODE_FIT
+            aspectRatioFrameLayout?.setAspectRatio(type)
+        }else if(type is Int) {
+            playerView?.resizeMode = type
```

Comment:
```suggestion
            playerView?.resizeMode = type
            aspectRatioFrameLayout?.setAspectRatio(0f) // Reset the aspect ratio when using built in resizing
```
Without this the implementation is bugged. This can easily be checked by switching from "Square" to "Fit to screen"

Reply:
Yes, I am also getting this while switching from Square to fit to screen, but after adding the setAspectRatio(0f) , after testing it was missing when switching from stretch to zoom, and other custom aspect ratio to predefined like fit, fill, zoom all behaving like fill. I am looking into that.

Reply:
Once I will try this on real device

Reply:
There maybe some issue in my code I will resolve it. I do agree with the idea setAspectRatio(0f) should fix the issue.

Reply:
@Akshay-1024 I cannot merge without a fix to this bug. Should I apply the fix or are you working on something?

Reply:
@fire-light42 I have removed fit as its working same as widescreen, I have removed Zoom as well because after the changes Zoom is basically behaving same as fill So Kept only Fill for predefined and rest all are custom aspect ratios rest all are working fine, but if you have any fix for zoom as well let me know. I have Pushed the changes

Reply:
I cant merge this if you have removed Fit and Zoom, all three original options act differently depending on screen size and video size. Just because they look the same on your phone, does not mean they function identically.